pandas/_libs/reduction.pyx

```diff
@@ -428,7 +432,9 @@ cdef class SeriesGrouper:
             vslider.reset()
 
         if result is None:
-            raise ValueError("No result.")
+            # This should no longer be reachable since we check for empty
+            #  series in the constructor.
+            raise AssertionError("`result` has not been assigned.")
```

Comment:
should this be just an assert statement. might be easier to grep.

Reply:
Yea agreed with @simonjayhawkins why not just `assert result is not None`?